File: data/src/main/java/com/arturogutierrez/openticator/storage/RealmDatabaseConfigurator.java
package com.arturogutierrez.openticator.storage;

import android.content.Context;
import android.util.Base64;
import com.arturogutierrez.openticator.domain.DatabaseConfigurator;
import io.realm.Realm;
import io.realm.RealmConfiguration;
import javax.inject.Inject;

public class RealmDatabaseConfigurator implements DatabaseConfigurator {

  private final Context context;

  @Inject
  public RealmDatabaseConfigurator(Context context) {
    this.context = context;
  }

  @Override
  public void configure(String passwordInBase64) {
    byte[] encryptionKey = Base64.decode(passwordInBase64, Base64.NO_WRAP);

    configureRealm(encryptionKey);
  }

  private void configureRealm(byte[] encryptionKey) {
    RealmConfiguration realmConfiguration =
        new RealmConfiguration.Builder(context).encryptionKey(encryptionKey).build();
    Realm.setDefaultConfiguration(realmConfiguration);
  }
}

File: data/src/main/kotlin/com/arturogutierrez/openticator/storage/RealmDatabaseConfigurator.kt
package com.arturogutierrez.openticator.storage

import android.content.Context
import android.util.Base64
import com.arturogutierrez.openticator.domain.DatabaseConfigurator
import io.realm.Realm
import io.realm.RealmConfiguration
import javax.inject.Inject

class RealmDatabaseConfigurator @Inject constructor(private val context: Context) : DatabaseConfigurator {

  override fun configure(passwordInBase64: String) {
    val encryptionKey = Base64.decode(passwordInBase64, Base64.NO_WRAP)

    configureRealm(encryptionKey)
  }

  private fun configureRealm(encryptionKey: ByteArray) {
    Realm.init(context)
    val realmConfiguration = RealmConfiguration.Builder().encryptionKey(encryptionKey).build()
    Realm.setDefaultConfiguration(realmConfiguration)
  }
}
